detox/src/devices/drivers/android/genycloud/GenyCloudDeviceAllocator.js

```diff
@@ -13,16 +13,18 @@ class GenyCloudDeviceAllocator extends AndroidDeviceAllocator {
 
   async _doAllocateDevice(recipe) {
     let { instance, isNew } = await this._doSynchronizedAllocation(recipe);
+    const { uuid, name } = instance;
+
     if (isNew) {
-      await this.deviceCleanupRegistry.allocateDevice(instance.uuid);
+      await this.deviceCleanupRegistry.allocateDevice({ uuid, name });
```

Comment:
Here the perversions start again 😃  — if you use specific-designed data classes for EmulatorHandle, GenymotionInstanceHandle with some equality or identity method implementations, you don't have to create on-fly these small objects.